test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/DefaultApplicationModelProviderTest.cs

```diff
@@ -116,6 +160,107 @@ public void OnProvidersExecuting_AddsBindingSources_ForActionParameters()
                 });
         }
 
+        [Fact]
+        public void OnProvidersExecuting_AddsBindingSources_ForActionParameters_WithLegacyValidationBehavior()
+        {
+            // Arrange
+            var modelMetadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var options = Options.Create(new MvcOptions { AllowValidatingTopLevelNodes = false });
```

Comment:
Why set `AllowValidatingTopLevelNodes = false` in some tests but use bare `new MvcOptions()` in others? Stick with one and I suggest that be bare `new MvcOptions()`.